Description: SUPLib-linux: new version number scheme starting with Linux 3.0
Author: upstream, http://www.virtualbox.org/changeset/37556

--- a/src/VBox/HostDrivers/Support/linux/SUPLib-linux.cpp
+++ b/src/VBox/HostDrivers/Support/linux/SUPLib-linux.cpp
@@ -267,9 +267,14 @@
         rc = RTStrToUInt32Ex(szBuf, &pszNext, 10, &uA);
         if (   RT_SUCCESS(rc)
-                && *pszNext == '.')
+            && *pszNext == '.')
         {
+            /*
+             * new version number scheme starting with Linux 3.0
+             */
+            if (uA >= 3)
+                return VINF_SUCCESS;
             rc = RTStrToUInt32Ex(pszNext+1, &pszNext, 10, &uB);
             if (   RT_SUCCESS(rc)
-                    && *pszNext == '.')
+                && *pszNext == '.')
             {
                 rc = RTStrToUInt32Ex(pszNext+1, &pszNext, 10, &uC);
